config/examples/core.yml

```diff
@@ -1,5 +1,5 @@
 base: &default
-  url: <%= ENV.fetch('THREESCALE_CORE_INTERNAL_API', 'http://localhost:3001/internal/') %>
+  url: "<%= URI.join(ENV.fetch('BACKEND_URL', 'http://localhost:3001'), '/internal/') %>"
```

Comment:
I would use `File.join` here
```rb
[1] pry(main)> URI.join(ENV.fetch('BACKEND_URL', 'http://localhost:3001'), '/internal/')
=> #<URI::HTTP http://localhost:3001/internal/>
[2] pry(main)> ENV["BACKEND_URL"] = 'http://localhost:3002'
=> "http://localhost:3002"
[3] pry(main)> URI.join(ENV.fetch('BACKEND_URL', 'http://localhost:3001'), '/internal/')
=> #<URI::HTTP http://localhost:3002/internal/>
[4] pry(main)> ENV["BACKEND_URL"] = 'http://localhost:3002/something'
=> "http://localhost:3002/something"
[5] pry(main)> URI.join(ENV.fetch('BACKEND_URL', 'http://localhost:3001'), '/internal/')
=> #<URI::HTTP http://localhost:3002/internal/>
```

Reply:
In practice probably it doesn't make any difference. But in case whoever wanted to deploy the backend API not to be at the root of the server, it would be a surprising effect.

Reply:
That's a great point! I hadn't considered this scenario, but I think it makes sense to allow the backend URL have some path!

Indeed, `URI.join` behavior is **extremely** confusing:

```
[1] pry(main)> URI.join('http://localhost:3001/something', 'internal')
=> #<URI::HTTP http://localhost:3001/internal>

[2] pry(main)> URI.join('http://localhost:3001/something/', 'internal')
=> #<URI::HTTP http://localhost:3001/something/internal>

[3] pry(main)> URI.join('http://localhost:3001/something/', '/internal')
=> #<URI::HTTP http://localhost:3001/internal>

[4] pry(main)> URI.join('http://localhost:3001/something', '/internal')
=> #<URI::HTTP http://localhost:3001/internal>
```

Using `File.join` for URLs might be a bit confusing because of the name, but it does the trick perfectly, so I'll change it as you suggested.